src/plugins/discover/public/application/components/data_grid/data_grid_table.tsx

```diff
@@ -208,7 +214,7 @@ export const DataGridTable = ({
         renderCellValue={renderCellValue}
         rowCount={rowCount}
         sorting={sorting}
-        toolbarVisibility={isToolbarVisible ? toolbarVisibility : false}
+        toolbarVisibility={isToolbarVisible ? toolbarOptions : false}
```

Comment:
The name of this prop gives an impression of being an object, and not a boolean value.

Reply:
It is an object. Thats why i think the OUI prop name here is actually pretty bad, but thats problem thats out of scope for this issue.